From 2f2f1748064ed2ff3a26b06005e28724780892e6 Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@edgedb.com>
Date: Tue, 26 Oct 2021 09:58:46 -0700
Subject: [PATCH] Use -Wl,-rpath on macOS also

The claim that macOS linker doesn't understand `-rpath` seems to be
outdated.
---
 Lib/distutils/unixccompiler.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Lib/distutils/unixccompiler.py b/Lib/distutils/unixccompiler.py
index f0792de74a..08440805b2 100644
--- a/Lib/distutils/unixccompiler.py
+++ b/Lib/distutils/unixccompiler.py
@@ -233,8 +233,7 @@ def runtime_library_dir_option(self, dir):
         # we use this hack.
         compiler = os.path.basename(sysconfig.get_config_var("CC"))
         if sys.platform[:6] == "darwin":
-            # MacOSX's linker doesn't understand the -R flag at all
-            return "-L" + dir
+            return "-Wl,-rpath," + dir
         elif sys.platform[:7] == "freebsd":
             return "-Wl,-rpath=" + dir
         elif sys.platform[:5] == "hp-ux":
-- 
2.32.0

